core/iwasm/libraries/lib-wasi-threads/test/build.sh

```diff
@@ -10,7 +10,7 @@ CC=${CC:=/opt/wasi-sdk/bin/clang}
 WAMR_DIR=../../../../..
 
 # Stress tests names
-thread_start_file_exclusions=("spawn_stress_test.wasm" "linear_memory_size_update.wasm")
+thread_start_file_exclusions=("spawn_stress_test.wasm" "linear_memory_size_update.wasm" "stress_test_threads_creation.wasm")
```

Comment:
for this new test we probably want to enable compiler optimizations so we can spawn as many threads as we can.

Reply:
that makes sense and I think we can safely set -O2 for all the tests here. Does it make sense to you?